source/routing/circle.cpp

```diff
@@ -0,0 +1,101 @@
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+Stay tuned using
+twitter @navitia
+IRC #navitia on freenode
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+
+#include "type/geographical_coord.h"
+#include "circle.h"
+#include "raptor.h"
+
+#include <set>
+#include <assert.h>
+#include <vector>
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/polygon.hpp>
+#include <cmath>
+
+namespace navitia { namespace routing {
+
+
+type::GeographicalCoord project_in_direction(const type::GeographicalCoord& center,
+                                             const double& direction,
+                                             const double& radius){
+    using navitia::type::GeographicalCoord;
+    assert(radius > 0);
+    double alpha = radius / GeographicalCoord::EARTH_RADIUS_IN_METERS;
+    double direction_rad = fmod(direction, 360) * GeographicalCoord::N_DEG_TO_RAD;
+    if (direction < 0) {
+        direction_rad = 2 * PI - direction_rad;
+    }
+    double center_lat_rad = center.lat() * GeographicalCoord::N_DEG_TO_RAD;
+    double center_lon_rad = center.lon() * GeographicalCoord::N_DEG_TO_RAD;
+    double delta_lat;
+    double delta_lon;
+    double lat;
+    double lon;
+    if (fmod(direction, 180) == 0) {
+        delta_lat = pow(-1, fmod(direction / 180, 2)) * alpha;
+        lat = center_lat_rad + delta_lat;
+        lon = center_lon_rad;
+    } else {
+        double projection = asin(sin(alpha) * sin(direction_rad));
+        delta_lat = acos(cos(alpha) / cos(projection));
+        if (direction_rad > PI / 2 && direction_rad < 3 * PI / 2) {
+            delta_lat = -1 * delta_lat;
+        }
+        lat = center_lat_rad + delta_lat;
+        delta_lon = acos((cos(alpha) - sin(lat) * sin(center_lat_rad)) / (cos(lat) * cos(center_lat_rad)));
+        if (direction_rad > PI) {
+            delta_lon = -1 * delta_lon;
+        }
+        lon = center_lon_rad + delta_lon;
+    }
+    double lon_deg = lon * N_RAD_TO_DEG;
+    double lat_deg = lat * N_RAD_TO_DEG;
+    //If latitude is not in  [-90,90] or longitude is not in [-180,180]
+    if (lat_deg < -90 || lat_deg > 90) {
+        lat_deg = (lat_deg / abs(lat_deg)) * 90 + fmod(lat_deg, 90);
+    }
+    if (lon_deg < -180 || lon_deg > 180) {
+        lon_deg = (lon_deg / abs(lon_deg)) * 180 + fmod(lon_deg, 180);
+    }
+    return type::GeographicalCoord(lon_deg, lat_deg);
+}
+
+
+type::Polygon circle(const type::GeographicalCoord& center,
+                     const double& radius) {
+    type::Polygon points;
+    for (double i = 0; i < 360; i++) {
+        points.outer().push_back(project_in_direction(center, i, radius));
```

Comment:
since you know the size of your vector, you can do a `reserve` or use `std::array`, a size fixed container